extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -124,7 +125,7 @@ public class MSQControllerTask extends AbstractTask implements ClientTaskQuery,
   @JsonCreator
   public MSQControllerTask(
       @JsonProperty("id") @Nullable String id,
-      @JsonProperty("spec") MSQSpec querySpec,
+      @JsonProperty("spec") LegacyMSQSpec querySpec,
```

Comment:
Since any changes to this class effect backward and forward compatibility and the changes to the legacy msq spec seem compatible, can we add a test case here were we take a few json payloads of `old msq spec` and try to convert them to `legacy msq spec` so that we donot accidentally break backward compatibility in the future.

Reply:
I was trying to be careful not to break it - but a test checking against serialized data is a good idea.
Added `MSQSpecCompatTest` with 3 cases